lisa/tools/hyperv.py

```diff
@@ -119,20 +120,21 @@ def create_vm(
                 force_run=True,
             )
 
-        # add disks
-        disk_info = powershell.run_cmdlet(
-            "(Get-Disk | Where-Object {$_.OperationalStatus -eq 'offline'}).Number",
-            force_run=True,
-        )
-        matched = re.findall(r"\d+", disk_info)
-        disk_numbers = [int(x) for x in matched]
-        for disk_number in disk_numbers:
-            self._run_hyperv_cmdlet(
-                "Add-VMHardDiskDrive",
-                f"-VMName {name} -DiskNumber {disk_number} -ControllerType 'SCSI'",
-                extra_args=extra_args,
+        # add disks if requested
+        if add_disks:
+            disk_info = powershell.run_cmdlet(
+                "(Get-Disk | Where-Object {$_.OperationalStatus -eq 'offline'}).Number",
                 force_run=True,
             )
+            matched = re.findall(r"\d+", disk_info)
+            disk_numbers = [int(x) for x in matched]
+            for disk_number in disk_numbers:
+                self._run_hyperv_cmdlet(
+                    "Add-VMHardDiskDrive",
+                    f"-VMName {name} -DiskNumber {disk_number} -ControllerType 'SCSI'",
```

Comment:
is the ControllerType always SCSI?

Reply:
This is existing code. I just moved it inside the `if` block.